draft-ietf-jsonpath-base.md

```diff
@@ -1502,10 +1502,24 @@ The well-typedness of function expressions can now be defined in terms of this t
 
 For a function expression to be well typed:
 
-1. its declared type must be well typed in the context in which it occurs, and
-2. its arguments must be well typed for the declared type of the corresponding parameters.
+1. its function name MUST be registered (see {{iana-fnex}}),
+2. its declared type must be well typed in the context in which it occurs, and
+3. its arguments must be well typed for the declared type of the corresponding parameters.
 
-(1) As per the grammar, a function expression can occur in three different
+(1) The registered set of function extensions is the source of knowledge regarding the declared return and
+parameter types of a function extension.
+
+If a function expression is treated as well typed, the function extensions involved in the expression belong
+to the registered set.
+However, an implementation may treat a function expression as not well typed even if all the function extensions
+involved in the expression are registered.
```

Comment:
Is this a roundabout way of saying that an implementation is allowed to not support a registered function?

Reply:
That's certainly an implication of that sentence, but there is another implication: the query must fail with an error.